OpenRA.Game/Traits/Health.cs

```diff
@@ -188,6 +188,15 @@ public static DamageState GetDamageState(this Actor self)
 			return (health == null) ? DamageState.Undamaged : health.DamageState;
 		}
 
+		public static bool IsDamaged(this Actor self)
```

Comment:
Why are you afraid of querying the DamageState? Both performance and rounding errors should not be a problem.